swap_meet/vendor.py

```diff
@@ -1,2 +1,101 @@
+from swap_meet.item import Item
+
+# Wave 1
+"""The class vendor takes in a parameter of inventory. It has a method function
+    that adds and removes items.
+    Returns:
+      It returns the item or the boolean false.
+        """
+
+
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+        self.item = Item()
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+# Wave 2
+        """This method checks through the items csategory and appends it to a list.
+        """
+
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+# Wave 3
+        """This item checks to see if an item is in my inventory or my friends
+        and it removes it from my items to append to theirs. The same is done on
+        their end.
+        """
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        if my_item in self.inventory:
+            self.remove(my_item)
+            vendor.add(my_item)
+        if their_item in vendor.inventory:
+            vendor.remove(their_item)
+            self.add(their_item)
+            return True
```

Comment:
Great use of a guard clause! This line already takes care of checking if `my_item` and `their_item` are truthy so we can omit the other conditionals:

```suggestion
        if my_item not in self.inventory or their_item not in vendor.inventory:
            return False
        
        self.remove(my_item)
        vendor.add(my_item)
        
        vendor.remove(their_item)
        self.add(their_item)
        return True
```